docs/protocol.md

```diff
@@ -275,10 +275,13 @@ new versions are required, this is where a common version can be negotiated.
 There are a variable number of key/value pairs. For version 2, the following
 are required:
 
-| name           | format             | description
-| :------------- | :----------------- | :----------
-| `host_port`    | `address:port`     | where this process can be reached
-| `process_name` | *arbitrary string* | additional identifier for this instance, used for logging
+| name                        | format             | description
+| :-------------------------- | :----------------- | :----------
+| `host_port`                 | `address:port`     | where this process can be reached
+| `process_name`              | *arbitrary string* | additional identifier for this instance, used for logging
+| `tchannel_language`         | *arbitrary string* | a short string naming the implementation language (e.g. `"node"`,` "python"`, `"go"`).
+| `tchannel_language_version` | *arbitrary string* | a language specific version string indicating the version of the runtime or build chain as appropriate (e.g. `"0.10.32", "2.7.10", "1.5.1")
```

Comment:
Fair question @breerly , I'm mostly trying to be conservative with the key names here:
- say in some future we have a C library version of tchannel, that we then use by bindings in other languages
- I'd rather be clear that "this is the language that the tchannel (at least core) code is written in", versus "this is the language that the service is written in"; now, we have no plans for such a world right now, but I'd rather keep the key named conservatively